packages/core/src/currentScopes.ts

```diff
@@ -74,15 +74,42 @@ export function withScope<T>(
  *
  * This function is intended for Sentry SDK and SDK integration development. It is not recommended to be used in "normal"
  * applications directly because it comes with pitfalls. Use at your own risk!
+ */
+export function withIsolationScope<T>(callback: (isolationScope: Scope) => T): T;
+/**
+ * Set the provided isolation scope as active in the given callback. If no
```

Comment:
we should mention in the jsdoc what happens if your pass in undefined as the isolationScope argument